### ISSUE TITLE
Upgrade frontend to elm 0.19

### ISSUE THREAD

**Issue by jhillyerd** [jhillyerd/inbucket#125]
SUCCESS! Your project's dependencies and code have been upgraded.
However, your project may not yet compile due to API changes in your
dependencies.

See <https://github.com/elm/compiler/blob/master/upgrade-docs/0.19.md>
and the documentation for your dependencies for more information.

WARNING! 1 of your dependencies have not yet been upgraded to
support Elm 0.19.
  - [x] https://github.com/elm-lang/websocket

Here are some common upgrade steps that you will need to do manually:

- NoRedInk/elm-json-decode-pipeline
  - [x] Changes uses of Json.Decode.Pipeline.decode to Json.Decode.succeed
- elm/core
  - [x] Replace uses of toString with String.fromInt, String.fromFloat, or Debug.toString as appropriate
- undefined
  - [x] Read the new documentation here: https://package.elm-lang.org/packages/elm/time/latest/
  - [x] Replace uses of Date and Time with Time.Posix
- elm/html
  - [x] If you used Html.program*, install elm/browser and switch to Browser.element or Browser.document
  - [x] If you used Html.beginnerProgram, install elm/browser and switch Browser.sandbox
- elm/browser
  - [x] Change code using Navigation.program* to use Browser.application
  - [x] Use the Browser.Key passed to your init function in any calls to Browser.Navigation.pushUrl/replaceUrl/back/forward
- elm/url
  - [x] Changes uses of Navigation.Location to Url.Url
  - [x] Change code using UrlParser.* to use Url.Parser.*

(output from https://github.com/avh4/elm-upgrade)

Remaining items:

- [x] Posix dates
- [x] Compile errors go to console, not hot-reload browser window
- [x] SPA routing broken in dev server
- [x] Update document title code for 0.19
- [x] External links do not open (github for example)
- [x] Monitor is not working well; document in new bug - #128 
- [x] Scan code for simple refactor opportunities
- [ ] Check-in npm lockfile

**Comment by jhillyerd** [comment]
Looks like this will be blocked on [websocket](https://github.com/elm-lang/websocket) availability for the time being.  I'd expect https://github.com/elm/websocket to exist when it's ready.

**Comment by jhillyerd** [comment]
Working websocket ports version now in `feature/ws-port` branch.

**Comment by jhillyerd** [comment]
WebSocket proxying doesn't seem work in the current version of create-elm-app, so I am crafting an artisan webpack recipe. :nauseated_face:

**Comment by jhillyerd** [comment]
ecd0c12 (ws-port) merged into develop branch.

I have a basic webpack config w/ dev server, proxy and hot reload working.

elm 0.19 broke my hashed SPA routing, so now is a good time to replace it with real routes.

**Comment by jhillyerd** [comment]
Experimented with uglifyjs, but with Elm's new optimization, after gzipping it only saved 1KB.  Not worth the risk/overhead.
